### PR TITLE
Fix all the little issues

### LINKED ISSUE #73: Deprecation warnings from astropy > 5
I just started over with a brand new anaconda environment today and saw these:

  /Users/jonathanpober/opt/anaconda3/envs/py39/lib/python3.9/site-packages/astropy/coordinates/baseframe.py:1183: AstropyDeprecationWarning: Transforming a frame instance to a frame class (as opposed to another frame instance) will not be supported in the future.  Either explicitly instantiate the target frame, or first convert the source frame instance to a `astropy.coordinates.SkyCoord` and use its `transform_to()` method.
    warnings.warn("Transforming a frame instance to a frame class (as opposed to another "

tests/test_observation.py::test_units
  /Users/jonathanpober/opt/anaconda3/envs/py39/lib/python3.9/site-packages/astropy/units/astrophys.py:154: AstropyDeprecationWarning: `littleh` is deprecated from module `astropy.units.astrophys` since astropy 5.0 and may be removed in a future version. Use `astropy.cosmology.units.littleh` instead.
    warnings.warn(

tests/test_sensitivity.py::test_track
  Transforming a frame instance to a frame class (as opposed to another frame instance) will not be supported in the future.  Either explicitly instantiate the target frame, or first convert the source frame instance to a `astropy.coordinates.SkyCoord` and use its `transform_to()` method.

tests/test_sensitivity.py::test_limited_k_range
  `littleh` is deprecated from module `astropy.units.astrophys` since astropy 5.0 and may be removed in a future version. Use `astropy.cosmology.units.littleh` instead.

### LINKED ISSUE #79: Link CLI tutorial in RTD
Right now it doesn't appear in the menu on the left

### LINKED ISSUE #80: All the little bits for RASG
@plaplant had the following comments that we should work through before merging into rasg-affiliates:

- [x] Why is the 21cmSense.egg-info directory tracked? Might need to clean up .gitignore to be more developer-friendly (e.g., exclude build dir, etc). 
- [x] Update the Author email under "CONTRIBUTORS.RST" to be current ;)
- [x] For `pip install .[test]`, it missed the `pytest-cov` dependency. 
- [x] Why is the numpy dependency pinned to be <1.22? Is there a way for this to be relaxed?
- [x] The docs should suggest to users how to run the tests!
- [x] Remove `test_observatory.py::test_projected_baselines`.
- [x] Remove `devel/`
- [x] Make sure intermediate files don't get dumped in CWD. 
- [x] Rendering on RTD does not handle math correctly. This should be updated.
- [x] I'm not a *huge* fan of using Pickle to save intermediate states of the calculation. Obviously this is very convenient, but it does not allow for reading anywhere outside of Python. One might imagine that this information could be more broadly applicable (and it has been!), and so using a more robust format like HDF5 is preferable.
- [x] No checking is done when using the CLI that the array file passed by the user is appropriate for the calculation. Not sure if this is a ~bug, but it does seem a little dangerous.
- [x] In _utils.py: the `phase` method states that it was borrowed from pyuvdata. Should we consider using the method directly?
- [x] In antpos.py: do we want to support a more HERA-like fractured configuration? Not necessary but would be nice.
- [x] In beam.py: it might help to have some documentation for how to define/provide a user-specified beam.
- [x] In cli.py: nothing is communicated back to the user if a plot was desired by matplotlib was not available. Perhaps at least echo out a message to that effect?
- [x] In conversions.py: allow user to specify desired cosmology? This is probably a very small effect, but still---better either to allow user to provide (if possible) or say, "Planck15, take it or leave it."
- [x] In conversions.py: X2Y docstring seems inconsistent for units of returned value. (dL_df converts the units to 1/MHz, but docstring claims 1/GHz).
- [x] There should be a clearer example of how the user is to specify their own power spectrum as a reference for the sensitivity calculation. It looks like from the comments in the sensitivity.yaml sample file that it's possible to supply it there, but it would be good to be more clear about this topic.
- [x] CLI tutorial is not showing up in the left-hand menu on RTD.
- [x] Regenerate all docs/tutorials with current version of the code.

### PR DESCRIPTION
**Pull request by steven-murray**
Fixes #80 
Fixes #73
Fixes #79

### CODE REVIEW

**Comment by steven-murray** [comment]
@jpober thank you, that is exactly the kind of thing we should do.

**Comment by mkolopanis** [comment]
can we set up read the docs to build this branch so we can check how it looks?

**Comment by steven-murray** [comment]
@mkolopanis it's there?

**Comment by mkolopanis** [comment]
oh i see it now, just need to look harder

File: py21cmsense/theory.py [hunk 1]
```diff
@@ -0,0 +1,152 @@
+"""A module defining an interface for theoretical predictions, eg. power spectra.
+
+This is simply so that one can calculate sample variance values for a particular
+theoretical model, with a uniform interface.
+
+To register a class as a theory power spectrum, simply decorate it with ``theorypower``.
```
Comment (mkolopanis) [inline py21cmsense/theory.py:6]: I'm a little confused by this statement, I only see the word "theorypower" used here in the repo. Is there something I'm missing?

**Review by mkolopanis — COMMENTED** [review]
This is looking really good I have a question though

File: py21cmsense/theory.py (hunk 1 above)
Comment (steven-murray) [reply py21cmsense/theory.py]: Ah, remnant from earlier code. It's updated now to correctly say to subclass from `TheoryModel`.

**Comment by steven-murray** [comment]
@mkolopanis and @jpober I think this is good to go now. It has a new tutorial that re-obtains Pober (2015) HERA memo results, and also it fixes all the comments in #80. This should be the last obstacle before being acceptable for RASG.

File: docs/conf.py [hunk 2]
```diff
@@ -34,35 +34,42 @@
...
+# html_sidebars = {
+#     "**": [
+#         "sidebar/scroll-start.html",
+#         "sidebar/brand.html",
+#         "sidebar/search.html",
+#         "sidebar/navigation.html",
+#         "sidebar/ethical-ads.html",
+#         "sidebar/scroll-end.html",
+#     ]
+# }
```
Comment (mkolopanis) [inline docs/conf.py:64]: are we keeping this around just in case or can we delete it?

File: docs/faqs.rst [hunk 3]
```diff
@@ -1,3 +1,128 @@
...
+sensitivity YAML configuration::
+
+    plugins:
+      my_theory.module
+
+    theory_model:
+      PowerLaw
+
+
+How do I change cosmology?
```
Comment (mkolopanis) [inline docs/faqs.rst:51]: you can't. we only have one observable universe.

**Review by mkolopanis — COMMENTED** [review]
I really only have on small question about some comments I saw

File: docs/conf.py (hunk 2 above)
Comment (steven-murray) [reply docs/conf.py:65]: I'm still working on it --- I realized the docs are not showing the sidebar all of a sudden. Will remove this entirely if it's not required.

Comment (mkolopanis) [reply docs/conf.py:65]: okie dokie :+1: wasn't sure if it was done done or not!

**Review by mkolopanis — APPROVED** [review]
Let's do it!

**Pull request merged**